### PR TITLE
Externalize JavaScript: more app manager

### PR DESCRIPTION
**Pull request by orangejenny**
review commits individually

@gcapalbo / @snopoke

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 15 errors:

* corehq/apps/app_manager/static/app_manager/js/apps_base.js, line 2 - 'COMMCAREHQ' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/apps_base.js, line 7 - Missing trailing comma. (comma-dangle)
* corehq/apps/app_manager/static/app_manager/js/apps_base.js, line 13 - 'hqLayout' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/apps_base.js, line 20 - 'hqLayout' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 4 - 'COMMCAREHQ' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 13 - 'langcodeTag' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 19 - Expected '===' and instead saw '=='. (eqeqeq)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 28 - Expected '===' and instead saw '=='. (eqeqeq)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 46 - Missing trailing comma. (comma-dangle)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 89 - 'gaTrackLink' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 92 - 'gaTrackLink' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 93 - 'gaTrackLink' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/value-or-none-ui.js, line 2 - 'bindingContext' is defined but never used (no-unused-vars)
* corehq/apps/app_manager/static/app_manager/js/value-or-none-ui.js, line 10 - 'ValueOrNoneUI' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/value-or-none-ui.js, line 17 - Missing trailing comma. (comma-dangle)

**Comment by orangejenny** [comment]
ignoring dimagimon

**Review by gcapalbo — APPROVED** [review]
looks ok to me, but i think build failure is related

File: corehq/apps/app_manager/templates/app_manager/v1/managed_app.html [hunk 1]
```diff
@@ -350,5 +258,11 @@
 {% endblock %}
 
 {% block app-content %}
+    {% initial_page_data 'form_counts_by_module' form_counts_by_module %}
+    {% initial_page_data 'app_version' app.version %}
+    {% initial_page_data 'app_commcare_minor_release' app.commcare_minor_release %}
+    {% initial_page_data 'latest_commcare_version' latest_commcare_version %}
+    {% initial_page_data 'app_doc_type' app.get_doc_type %}
+    {% initial_page_data 'formdesigner' formdesigner %}
```
Comment (snopoke) [inline corehq/apps/app_manager/templates/app_manager/v1/managed_app.html]: Is there a reason you didn't group these into a single object?

**Review by snopoke — APPROVED** [review]

**Comment by dimagimon** [comment]
There are 15 errors:

* corehq/apps/app_manager/static/app_manager/js/apps_base.js, line 2 - 'COMMCAREHQ' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/apps_base.js, line 7 - Missing trailing comma. (comma-dangle)
* corehq/apps/app_manager/static/app_manager/js/apps_base.js, line 13 - 'hqLayout' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/apps_base.js, line 20 - 'hqLayout' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 4 - 'COMMCAREHQ' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 13 - 'langcodeTag' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 19 - Expected '===' and instead saw '=='. (eqeqeq)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 28 - Expected '===' and instead saw '=='. (eqeqeq)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 46 - Missing trailing comma. (comma-dangle)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 89 - 'gaTrackLink' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 92 - 'gaTrackLink' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 93 - 'gaTrackLink' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/value-or-none-ui.js, line 2 - 'bindingContext' is defined but never used (no-unused-vars)
* corehq/apps/app_manager/static/app_manager/js/value-or-none-ui.js, line 10 - 'ValueOrNoneUI' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/value-or-none-ui.js, line 17 - Missing trailing comma. (comma-dangle)

File: corehq/apps/app_manager/templates/app_manager/v1/managed_app.html (hunk 1 above)
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/v1/managed_app.html]: Didn't notice how many of these `initial_page_data` tags there were. Good point, will update.

**Comment by dimagimon** [comment]
There are 14 errors:

* corehq/apps/app_manager/static/app_manager/js/apps_base.js, line 2 - 'COMMCAREHQ' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/apps_base.js, line 7 - Missing trailing comma. (comma-dangle)
* corehq/apps/app_manager/static/app_manager/js/apps_base.js, line 13 - 'hqLayout' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/apps_base.js, line 20 - 'hqLayout' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 4 - 'COMMCAREHQ' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 14 - 'langcodeTag' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 20 - Expected '===' and instead saw '=='. (eqeqeq)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 47 - Missing trailing comma. (comma-dangle)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 90 - 'gaTrackLink' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 93 - 'gaTrackLink' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/managed_app.js, line 94 - 'gaTrackLink' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/value-or-none-ui.js, line 2 - 'bindingContext' is defined but never used (no-unused-vars)
* corehq/apps/app_manager/static/app_manager/js/value-or-none-ui.js, line 10 - 'ValueOrNoneUI' is not defined. (no-undef)
* corehq/apps/app_manager/static/app_manager/js/value-or-none-ui.js, line 17 - Missing trailing comma. (comma-dangle)

File: corehq/apps/app_manager/templates/app_manager/v1/managed_app.html (hunk 1 above)
Comment (orangejenny) [inline corehq/apps/app_manager/templates/app_manager/v1/managed_app.html]: https://github.com/dimagi/commcare-hq/pull/14875/commits/d5d5a7f560d17d5d76cefe6e4610f3daefc99b66

**Pull request merged**